examples/aligned_accessor/aligned_accessor.cpp

```diff
@@ -0,0 +1,737 @@
+{
+  TICK();
+  for (std::size_t trial = 0; trial < num_trials; ++trial) {
+    add_omp_aligned_simd_raw_1d(n, x, y, z, ba);
+  }
+  return TOCK();
+}
+
+template<class ElementType, std::size_t byte_alignment>
+void add_omp_simd_aligned_raw_1d(const index_type n,
+				 aligned_pointer_t<const ElementType, byte_alignment> x,
+				 aligned_pointer_t<const ElementType, byte_alignment> y,
+				 aligned_pointer_t<ElementType, byte_alignment> z)
+{
+#pragma omp simd
+  for (index_type i = 0; i < n; ++i) {
+    z[i] = x[i] + y[i];
+  }
+}
+
+// Assume that x, y, and z all have the same alignment.
+template<class ElementType, std::size_t byte_alignment>
+auto benchmark_add_omp_simd_aligned_raw_1d(const std::size_t num_trials,
+					   const index_type n,
+					   const ElementType x[],
+					   const ElementType y[],
+					   ElementType z[],
+					   std::integral_constant<std::size_t, byte_alignment> ba)
+{
+  TICK();
+  auto x_blessed = bless(x, ba);
+  auto y_blessed = bless(y, ba);
+  auto z_blessed = bless(z, ba);
+  for (std::size_t trial = 0; trial < num_trials; ++trial) {
+    add_omp_simd_aligned_raw_1d<ElementType, byte_alignment>(n, x_blessed, y_blessed, z_blessed);
+  }
+  return TOCK();
+}
+
+template<class ElementType, std::size_t byte_alignment>
+void add_omp_aligned_simd_aligned_raw_1d(const index_type n,
+					 aligned_pointer_t<const ElementType, byte_alignment> x,
+					 aligned_pointer_t<const ElementType, byte_alignment> y,
+					 aligned_pointer_t<ElementType, byte_alignment> z)
+{
+#pragma omp simd aligned(z,x,y:byte_alignment)
+  for (index_type i = 0; i < n; ++i) {
+    z[i] = x[i] + y[i];
+  }
+}
+
+// Assume that x, y, and z all have the same alignment.
+template<class ElementType, std::size_t byte_alignment>
+auto benchmark_add_omp_aligned_simd_aligned_raw_1d(
+  const std::size_t num_trials,
+  const index_type n,
+  const ElementType x[],
+  const ElementType y[],
+  ElementType z[],
+  std::integral_constant<std::size_t, byte_alignment> ba)
+{
+  TICK();
+  auto x_blessed = bless(x, ba);
+  auto y_blessed = bless(y, ba);
+  auto z_blessed = bless(z, ba);
+  for (std::size_t trial = 0; trial < num_trials; ++trial) {
+    // Passing in ba doesn't help the compiler
+    // deduce the template parameters,
+    // so we just specify them explicitly.
+    add_omp_aligned_simd_aligned_raw_1d<ElementType, byte_alignment>(n, x_blessed, y_blessed, z_blessed);
+  }
+  return TOCK();
+}
+#endif // _OPENMP
+
+template<class ElementType>
+auto warmup(const std::size_t num_trials,
+	    const index_type n,
+	    const ElementType x[],
+	    const ElementType y[],
+	    ElementType z[])
+{
+  return benchmark_add_raw_1d(num_trials, n, x, y, z);
+}
+
+} // namespace (anonymous)
+
+int main(int argc, char* argv[])
+{
+  using std::cout;
+  using std::cerr;
+  using std::endl;
+
+  if(argc != 3) {
+    cerr << "Usage: main <n> <num_trials>" << endl;
+    return -1;
+  }
+
+  int n = 10000;
+  int num_trials = 100;
```

Comment:
Why do you bother with initializing to overwrite the value below

Reply:
@dalg24 Fixed; thanks!